src/errors.rs

```diff
@@ -18,6 +18,12 @@ impl std::fmt::Display for ParseError {
     }
 }
 
+impl From<std::convert::Infallible> for ParseError {
+    fn from(_: std::convert::Infallible) -> Self {
+        unreachable!()
+    }
+}
```

Comment:
Where is it actually used?

Reply:
Good question. A quick test build says "this is unused".

Reply:
It's used in xinput. It could probably be eliminated with some more special casing in the code generator.

Reply:
I forgot to actually enable the necessary features in my test build. :facepalm: 
New try says:
```
error[E0271]: type mismatch resolving `<u8 as std::convert::TryFrom<u8>>::Error == errors::ParseError`
    --> src/protocol/xinput.rs:8798:29
     |
8798 |         let format = format.try_into()?;
     |                             ^^^^^^^^ expected enum `std::convert::Infallible`, found enum `errors::ParseError`

error[E0271]: type mismatch resolving `<u8 as std::convert::TryFrom<u8>>::Error == errors::ParseError`
     --> src/protocol/xinput.rs:13546:29
      |
13546 |         let format = format.try_into()?;
      |                             ^^^^^^^^ expected enum `std::convert::Infallible`, found enum `errors::ParseError`

error: aborting due to 2 previous errors
```
This is inside `ChangeDevicePropertyRequest` and `XIChangePropertyRequest` In both cases, a `format` `u8` is converted to `u8`.

Reply:
Yeah we could probably try harder to avoid that u8 -> u8 conversion if we wanted to.

Reply:
I believe that this `try_into` comes from `emit_field_post_parse`, which would normally convert it into an enum. But here it is "converted" into an `u8` because it is a deducible field.

In `emit_struct_type`, this is avoided with:
```rust
if !field
    .name()
    .map(|field_name| deducible_fields.contains_key(field_name))
    .unwrap_or(false)
{
    self.emit_field_post_parse(field, out);
}
```